src/script/lua_api/l_particles_local.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+Copyright (C) 2017 red-001 <red-001@outlook.ie>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "lua_api/l_particles_local.h"
+#include "common/c_content.h"
+#include "common/c_converter.h"
+#include "lua_api/l_internal.h"
+#include "lua_api/l_object.h"
+#include "particles.h"
+
+// add_particle({pos=, velocity=, acceleration=, expirationtime=,
+// 		size=, collisiondetection=, collision_removal=, vertical=,
+//		texture=})
+// pos/velocity/acceleration = {x=num, y=num, z=num}
+// expirationtime = num (seconds)
+// size = num
+// collisiondetection = bool
+// collision_removal = bool
+// vertical = bool
+// texture = e.g."default_wood.png"
+// animation = TileAnimation definition
+// glow = num
```

Comment:
Types should  be embedded into line 28-30 or removed entirely (we have the documentation in `lua_api.txt`)